pkg/objects/yum.go

```diff
@@ -99,45 +98,43 @@ func (s *YumSource) ListAllFiles() ([]FileSpec, *ListEntriesError) {
 	}
 	repomdBytes, repomdURL, lerr := s.downloadAndParseXML(repomdPath, &repomd, cache)
 	if lerr != nil {
-		return nil, lerr
+		return lerr
 	}
 
-	signaturePath := repomdPath + ".asc"
 	//verify repomd's GPG signature
 	if s.gpgVerification {
+		signaturePath := repomdPath + ".asc"
 		signatureBytes, signatureURI, lerr := s.urlSource.getFileContents(signaturePath, cache)
 		if lerr == nil {
 			err := util.VerifyDetachedGPGSignature(s.gpgKeyRing, repomdBytes, signatureBytes)
 			if err != nil {
 				logg.Debug("could not verify GPG signature at %s for file %s", signatureURI, "-"+filepath.Base(repomdPath))
-				return nil, &ListEntriesError{
+				return &ListEntriesError{
 					Location: s.urlSource.getURLForPath("/").String(),
 					Message:  ErrMessageGPGVerificationFailed,
 					Inner:    err,
 				}
 			}
-			allFiles = append(allFiles, signaturePath)
+			out <- getFileSpec(signaturePath, cache)
 		} else {
 			if !strings.Contains(lerr.Message, "GET returned status 404") {
-				return nil, lerr
+				return lerr
 			}
 		}
 		logg.Debug("successfully verified GPG signature at %s for file %s", signatureURI, "-"+filepath.Base(repomdPath))
-	} else {
-		allFiles = append(allFiles, signaturePath)
```

Comment:
You're reverting 9f177db3c3a8a27bb6d5075b3958c37fab39b9d5 here. Please don't do that. This could use a comment why we're explicitly transferring the signature file when signature verification is being skipped.